### PR TITLE
Replace generic Dict calls in favour of Map

### PR DESCRIPTION
**Pull request by andreaazzini**
Closes https://github.com/hassox/guardian/issues/26.

### CODE REVIEW

File: lib/guardian/channel.ex [hunk 1]
```diff
@@ -36,7 +36,7 @@ defmodule Guardian.Channel do
   """
   defmacro __using__(opts) do
     opts = Enum.into(opts, %{})
-    key = Dict.get(opts, :key, :default)
+    key = Map.get(opts, :key, :default)
```
Comment (hassox) [inline lib/guardian/channel.ex]: This may not be a map so should stay dict.

File: lib/guardian/plug/ensure_permissions.ex [hunk 2]
```diff
@@ -13,16 +13,16 @@ defmodule Guardian.Plug.EnsurePermissions do
   """
   def init(opts) do
     opts = Enum.into(opts, %{})
-    on_failure = Dict.get(opts, :on_failure)
-    key = Dict.get(opts, :key, :default)
-    perms = Dict.delete(opts, :on_failure) |> Dict.delete(:key)
+    on_failure = Map.get(opts, :on_failure)
```
Comment (hassox) [inline lib/guardian/plug/ensure_permissions.ex]: May not be a map. This should stay a Dict.

File: lib/guardian/plug/ensure_authenticated.ex [hunk 3]
```diff
@@ -15,7 +15,7 @@ defmodule Guardian.Plug.EnsureAuthenticated do
   @doc false
   def init(opts) do
     opts = Enum.into(opts, %{})
-    case Dict.get(opts, :on_failure) do
+    case Map.get(opts, :on_failure) do
```
Comment (hassox) [inline lib/guardian/plug/ensure_authenticated.ex]: May not be a map. Should stay a Dict.

File: lib/guardian/plug.ex [hunk 4]
```diff
@@ -50,8 +50,8 @@ defmodule Guardian.Plug do
   """
   @spec sign_in(Plug.Conn.t, any, atom | String.t, Map) :: Plug.Conn.t
   def sign_in(conn, object, type, claims) do
-    the_key = Dict.get(claims, :key, :default)
-    claims = Dict.delete(claims, :key)
+    the_key = Map.get(claims, :key, :default)
```
Comment (hassox) [inline lib/guardian/plug.ex]: This may not be a map. This should stay a Dict.

File: lib/guardian/utils.ex [hunk 5]
```diff
@@ -3,7 +3,7 @@ defmodule Guardian.Utils do
   @doc false
   def stringify_keys(nil), do: %{}
   def stringify_keys(map) do
-    Enum.reduce(Dict.keys(map), %{}, fn(k,acc) -> Dict.put(acc, to_string(k), map[k]) end)
+    Enum.reduce(Map.keys(map), %{}, fn(k,acc) -> Map.put(acc, to_string(k), map[k]) end)
```
Comment (hassox) [inline lib/guardian/utils.ex]: This first one may not be a map. The one inside the anon function is for sure, but the other... not so sure.

**Comment by hassox** [comment]
LGTM. Just a couple of places where there could be either a map or KW list.

**Comment by andreaazzini** [comment]
I think the problem with those two places reside in the nature of `stringify_keys` itself. If we handled its input data so that it is not `nil` in the first place, we'd be completely fine using `Map`. What do you think about this?

**Comment by hassox** [comment]
The stringify keys could be getting a keyword list from arguments. If we make sure that those places are all good I think we should be ok. We'll just have to Enum.into a map before calling stringify keys.

File: lib/guardian/utils.ex (hunk 5 above)
Comment (lexmag) [inline lib/guardian/utils.ex]: I think for this case `Enum.reduce(map, %{}, fn({key, val},acc) -> Map.put(acc, to_string(key), val) end)` would be way better.

Comment (andreaazzini) [inline lib/guardian/utils.ex]: This looks good to me. @hassox any thought on this?

**Comment by hassox** [comment]
This is aging. I've been cleaning the Dict calls up as I come across them. Please update this PR if you want me to bring it in. :+1:

**Comment by hassox** [comment]
sorted in https://github.com/ueberauth/guardian/pull/77

**Pull request closed without merging**